packages/light.js/src/frequency/blocks.ts

```diff
@@ -20,11 +20,16 @@ import { onSyncingChanged$ } from './health';
  */
 const onEveryBlockWithApi$ = memoizee(
   (api: any, options?: FrequencyObservableOptions) =>
-    createPubsubObservable('eth_blockNumber', options).pipe(
-      withLatestFrom(onSyncingChanged$(options)),
+    createPubsubObservable(
+      'eth_newHeads',
+      'eth_getBlockByNumber',
+      options
+    ).pipe(
+      withLatestFrom(onSyncingChanged$(options).pipe(startWith(false))),
```

Comment:
I might be wrong but wouldn't this make onEveryBlockWithApi$ emit on initialization, even though the user is not synced? e.g.:
- onEveryBlockWithApi$ is called
- blockNumber RPC answers
- onEveryBlockWithApi$ emits wit the blockNumber
- isSyncing RPC answers `true` (which was always the case)
- onEveryBlockWithApi$ doesn't emit until isSyncing RPC answers `false`

Reply:
I imagine a fix could be to make onSyncingChanged$ fire `eth_syncing` (non-pubsub rpc call) first to get an initial value, and then switch to the pubsub. non-pubsub "eth_syncing" does get an answer.

obviously not very clean though

Reply:
I actually think it's a good idea, I thought about it too.

Another reason why it's good: `parity_subscribe('eth_getBlock')` immediately gives you an answer, and then gives you subsequent pubsub answers on each pub. `eth_subscribe('newHeads')` however doesn't give you the initial one, so there might be a (up to 15s) delay between subscribe and first event fired.

Reply:
ah, makes sense then!

Reply:
https://github.com/paritytech/js-libs/issues/61